RELEASE.md

```diff
@@ -0,0 +1,101 @@
+# Releases
+
+## Prerequisites - what you need to release WooCommerce Blocks
+- Install & set up [GitHub hub](https://hub.github.com) tools.
+- Configure a GitHub token for release scripts to use.
+  - https://github.com/settings/tokens
+  - Select the following permissions: 
+    - `admin:org`
+    - `public_repo`
+    - `read:user`
+  - Ensure your token is available to scripts, e.g. `export GH_API_TOKEN={YOUR-TOKEN}` or similar.
+- Get WPORG plugin svn commit access - ask a team member.
+
+_Outcome_: __You are equipped to ship a release!__
+
+## Release process
+### Lead-up to release
+#### Ensure release development is complete
+- Release milestone should exist (!).
+- Ensure all issues/PRs intended for this release are merged, closed and assigned to release milestone.
+- All PRs should have changelog entry, or `skip-changelog` tag.
+- Check with team to confirm any outstanding or in progress work.
+
+Note: changelog should be formatted like this in PR description. Note the preceding `> ` - this is required by changelog script.
+
+```md
+### Changelog
+
+> bug: Fix bug in Safari and other Webkit browsers that was causing the All Products block to show 0 results when resetting the sort value.
+```
+
+_Outcome_: __Team is aware of release and in agreement about what fixes & features are included.__
+
+#### Ensure release branch includes all relevant fixes
+- Make release branch if needed (major/minor).
+  - For fix releases, the branch should already exist.
+  - Otherwise create branch: `release/X.X`.
+- Update your local checkout to tip of the release branch.
+- Cherry pick all PRs into the release branch:
+  - If branch is already labelled `status: cherry-picked 🍒` then continue to next PR.
+  - Ideally, use GitHub Hub to cherry pick the PR - `hub am -3 {http://URL-TO-PR}`. 
+  - If there are serious conflicts or extensive differences between `master` and release branch, you may need to take more care:
+    - Manually cherry pick individual commits using git - `git cherry-pick {COMMIT-HASH}`.
+    - Or in some cases, manually craft a new PR with appropriate changes, targeting release branch.
+  - Push the release branch to origin (so changes are in GitHub repo).
+  - Label the PR: `status: cherry-picked 🍒`.
+
+_Outcome_: __Release branch has all relevant changes merged & pushed.__
+
+### Prepare release
+#### Ensure release branch readme is up to date
+- Run changelog script `npm run changelog` to get changelog txt for readme. Changelog content will be output to screen by script.
```

Comment:
This will need updated once @senadir or myself improve the changelog to allow for correctly selecting either Zenhub releases or Github milestones as the source for changelog entries. More than likely we'll have this configured to use cli flags and update the scripts in `package.json` so we'd have `npm run changelog:zh` and `npm rum changelog:gh` variations. I'm just noting this for now, we could publish as is and just update when we get the changelog update done.

Reply:
Thanks for your vigilance! We can update this later. This is a living doc, this is just the first iteration.

Note also from my point of view, the "cheatsheet" info (actual commands) is less important to document, compared to the phases and intentions/outcomes of the release steps. 

As we road-test the document on more releases, and/or as our tooling evolves, some of these details may be hard to maintain and get removed – I think this is ok. If the outcomes and phases are relatively stable then the doc is doing the job IMO 🦄.